packages/firebase/externs/firebase-firestore-externs.js

```diff
@@ -712,42 +714,45 @@ firebase.firestore.SnapshotMetadata.prototype.isEqual = function(other) {};
  * access will return 'undefined'. You can use the `exists` property to
  * explicitly verify a document's existence.
  *
- * @interface
+ * @constructor
```

Comment:
Why did this change?  We don't allow creating DocumentSnapshot objects directly...  and I see we (successfully?) use `@constructor` elsewhere in this file...

Reply:
woops, I meant we successfully use `@interface` elsewhere in this file.

Reply:
The closure compiler error was "constructors can only extend constructors" when `QueryDocumentSnapshot` tried to extend this.

Reply:
I see.  Perhaps the better fix is to change QueryDocumentSnapshot to use`@interface` then.  I don't care super strongly about these externs being perfectly correct though. :)